cheddar/cheddar-integration-aws/src/test/java/com/clicktravel/infrastructure/persistence/aws/dynamodb/DynamoDbTemplateTest.java

```diff
@@ -1384,6 +1384,42 @@ public void shouldDeleteItem_withStubItem() throws Exception {
     }
 
     @Test
+    public void shouldNotDeleteItem_withConditionalCheckFailedException() throws Exception {
+        // Given
+        final StubItem stubItem = new StubItem();
+        stubItem.setId(randomId());
+        stubItem.setVersion(randomLong());
+        final ItemConfiguration itemConfiguration = new ItemConfiguration(StubItem.class, tableName);
+        final Collection<ItemConfiguration> itemConfigurations = Arrays.asList(itemConfiguration);
+        when(mockDatabaseSchemaHolder.itemConfigurations()).thenReturn(itemConfigurations);
+        when(mockAmazonDynamoDbClient.deleteItem(any(DeleteItemRequest.class)))
+                .thenThrow(ConditionalCheckFailedException.class);
+
+        final DynamoDbTemplate dynamoDbTemplate = new DynamoDbTemplate(mockDatabaseSchemaHolder);
+        dynamoDbTemplate.initialize(mockAmazonDynamoDbClient);
+
+        // When
+        OptimisticLockException actualException = null;
+        try {
+            dynamoDbTemplate.delete(stubItem);
+        } catch (final OptimisticLockException e) {
+            actualException = e;
+        }
+
+        // Then
+        assertNotNull(actualException);
+        final ArgumentCaptor<DeleteItemRequest> deleteItemRequestArgumentCaptor = ArgumentCaptor
+                .forClass(DeleteItemRequest.class);
+        verify(mockAmazonDynamoDbClient).deleteItem(deleteItemRequestArgumentCaptor.capture());
```

Comment:
Need to ensure this is only called once i.e. none of the constraints are deleted.